src/ttl_table.c

```diff
@@ -132,3 +132,170 @@ bool TimeToLiveTable_VerifyDocAndFields(TimeToLiveTable *table, t_docId docId, c
   }
   return false;
 }
+
+bool TimeToLiveTable_VerifyDocAndField(TimeToLiveTable *table, t_docId docId, t_fieldIndex field, enum FieldExpirationPredicate predicate, const struct timespec* expirationPoint) {
+  dictEntry *entry = dictFind(table, (void*)docId);
+  if (!entry) {
```

Comment:
Just to have common logic, couldn't we just create a FieldMask with the field bit to 1 and call the generic version?

Reply:
because the setup is a bit complex (`t_fieldIndex` is not necessarily the bit offset in `t_fieldMask`) and I prefer to have a dedicated logic for this case, as for the iterator over this inverted index will always has the same field input and may call this function for each `Read` and `SkipTo`, potentially more than once per API call